targets/TARGET_NORDIC/TARGET_NRF5x/nrf_log.h

```diff
@@ -101,16 +96,34 @@ extern "C" {
  *
  *  @details This macro is compiled only if @ref NRF_LOG_LEVEL includes debug logs.
  */
-
-#define NRF_LOG_ERROR(...)                     NRF_LOG_INTERNAL_ERROR(__VA_ARGS__)
-#define NRF_LOG_WARNING(...)                   NRF_LOG_INTERNAL_WARNING( __VA_ARGS__)
-#define NRF_LOG_INFO(...)                      NRF_LOG_INTERNAL_INFO( __VA_ARGS__)
-#define NRF_LOG_DEBUG(...)                     NRF_LOG_INTERNAL_DEBUG( __VA_ARGS__)
+#define NRF_LOG_DEBUG(x, ...)                                        \
+    if (NRF_LOG_ENABLED && (NRF_LOG_LEVEL >= NRF_LOG_LEVEL_DEBUG))   \
+    {                                                                \
+            printf("[NRF_DBG] "x"\r\n", ##__VA_ARGS__);              \
```

Comment:
I don't think this is portable (won't work if you do `NRF_LOG_DEBUG("I don't have parameters");`

Reply:
Oh, don't the other compilers handle these kind of optional macro arguments? They really are substandard for the money you pay for them :-(